services/QuillLMS/app/workers/user_login_worker.rb

```diff
@@ -16,6 +16,7 @@ def perform(id)
         SegmentIo::BackgroundEvents::TEACHER_SIGNIN,
         properties: @user&.segment_user&.common_params
       )
+      record_user_login
```

Comment:
What about a method in user.rb:

```
def record_login
  user_logins.create
end
```

Then the we're left with `@user.record_login` which removes the dependency on `UserLogin`

Reply:
I did consider that.  There's a part of me that's become quite resistant to adding more code to `User` because that model is HUGE and complex and I don't want to add to it.  Am I just being silly with that since it really does make the most sense to have that method in the model?